azure-ipam/logger/logger.go

```diff
@@ -6,56 +6,48 @@ import (
 	"github.com/pkg/errors"
 	"go.uber.org/zap"
 	"go.uber.org/zap/zapcore"
+	"gopkg.in/natefinch/lumberjack.v2"
 )
 
 type Config struct {
-	Level            string
-	OutputPaths      string // comma separated list of paths
-	ErrorOutputPaths string // comma separated list of paths
+	Level       string // Debug by default
+	Filename   	string
+	MaxSize    	int    // megabytes
+	MaxBackups 	int    // # of backups, no limitation by default
+	MaxAge     	int    // days, no limitation by default
 }
 
 // NewLogger creates and returns a zap logger and a clean up function
 func New(cfg *Config) (*zap.Logger, func(), error) {
-	loggerCfg := &zap.Config{}
-
-	level, err := zapcore.ParseLevel(cfg.Level)
-	if err != nil {
-		return nil, nil, errors.Wrapf(err, "failed to parse log level")
-	}
-	loggerCfg.Level = zap.NewAtomicLevelAt(level)
-
-	loggerCfg.Encoding = "json"
-	loggerCfg.OutputPaths = getLogOutputPath(cfg.OutputPaths)
-	loggerCfg.ErrorOutputPaths = getErrOutputPath(cfg.ErrorOutputPaths)
-	loggerCfg.EncoderConfig = zapcore.EncoderConfig{
-		TimeKey:     "time",
-		MessageKey:  "msg",
-		LevelKey:    "level",
-		EncodeLevel: zapcore.LowercaseLevelEncoder,
-		EncodeTime:  zapcore.ISO8601TimeEncoder,
-	}
-
-	logger, err := loggerCfg.Build()
-	if err != nil {
+	//check the filepath is not empty
```

Comment:
the linter is being lenient for you... comments must have a space after the `//`. (and they should explain _why_, not what, you are doing).

Reply:
will address in the next commit